apache/zeppelin
# [ZEPPELIN-1999][IMPROVEMENT] refactoring replacement context parameters into interpreter settings

### What is this PR for?
Field `protected Properties property` is protected. sometimes this field used derictly. Sometimes  interpreter is called from another insterpreter (for ex. Spark, SparkInterpreter creates spark context and calls PySpark). This improvement fix it problems with context parameters replacement into properties.

### What type of PR is it?
Improvement


### How should this be tested?
1) Setting Spark interpreter
   set property 'spark.app.name=#{noteId}'
2) Run paragraph
  ```%pyspark 
      print(sc._conf.get("spark.app.name"))
```
3) Check result (note id)
4) Open http://localhost:4040/environment/ (spark ui), check property `spark.app.name` = note id


### Screenshots (if appropriate)

### Questions:
* Does the licenses files need update? no
* Is there breaking changes for older versions? no
* Does this needs documentation? no
